runelite-client/src/main/java/net/runelite/client/plugins/poison/PoisonPlugin.java

```diff
@@ -0,0 +1,253 @@
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.poison;
+
+import com.google.inject.Provides;
+import java.awt.Color;
+import java.awt.FontMetrics;
+import java.awt.Graphics;
+import java.awt.image.BufferedImage;
+import java.text.MessageFormat;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import javax.inject.Inject;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Client;
+import net.runelite.api.SpriteID;
+import net.runelite.api.VarPlayer;
+import net.runelite.api.events.ConfigChanged;
+import net.runelite.api.events.VarbitChanged;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.game.SpriteManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.FontManager;
+import net.runelite.client.ui.overlay.OverlayManager;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+import net.runelite.client.util.ColorUtil;
+
+@PluginDescriptor(
+	name = "Poison",
+	description = "Tracks current damage values for Poison and Venom",
+	tags = {"combat", "poison", "venom"}
+)
+@Slf4j
+public class PoisonPlugin extends Plugin
+{
+	static final int POISON_TICK_MILLIS = 18200;
+	private static final int VENOM_THRESHOLD = 1000000;
+	private static final int VENOM_MAXIUMUM_DAMAGE = 20;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private PoisonOverlay poisonOverlay;
+
+	@Inject
+	private OverlayManager overlayManager;
+
+	@Inject
+	private InfoBoxManager infoBoxManager;
+
+	@Inject
+	private SpriteManager spriteManager;
+
+	@Inject
+	private PoisonConfig config;
+
+	@Getter
+	private int lastDamage;
+	private boolean envenomed;
+	private PoisonInfobox infobox;
+	private Instant poisonNaturalCure;
+	private Instant nextPoisonTick;
+	private int lastValue = -1;
+
+	@Provides
+	PoisonConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(PoisonConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		overlayManager.add(poisonOverlay);
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		overlayManager.remove(poisonOverlay);
+
+		if (infobox != null)
+		{
+			infoBoxManager.removeInfoBox(infobox);
+			infobox = null;
+		}
+
+		envenomed = false;
+		lastDamage = 0;
+		poisonNaturalCure = null;
+		nextPoisonTick = null;
+		lastValue = -1;
+
+	}
+
+	@Subscribe
+	public void onVarbitChanged(VarbitChanged event)
+	{
+		final int poisonValue = client.getVar(VarPlayer.POISON);
+		if (poisonValue == lastValue)
+		{
+			return;
+		}
+
+		lastValue = poisonValue;
+		nextPoisonTick = Instant.now().plus(Duration.of(POISON_TICK_MILLIS, ChronoUnit.MILLIS));
```

Comment:
Drinking an anti potion or getting repoisoned changes the var but doesn't change when the next poison tick will happen.

This makes the first hitsplat after getting repoisoned/venomed sooner than expected. A check if the change is 30 ticks after the last poison tick would fix it.